lib/model/message.dart

```diff
@@ -168,10 +181,26 @@ class MessageStoreImpl with MessageStore {
       return;
     }
 
-    // final newStreamId = event.newStreamId; // null if topic-only move
+    final newChannelId = event.newStreamId; // null if topic-only move
+
+    // Eliminate the case when the topic is updated only to be resolved/unresolved.
+    // return early as the messages' editState need no update.
+    if (
+      newChannelId == null
+      && MessageEditState.areSameTopic(event.origTopic!, event.newTopic!)
+    ) return;
+
     // final newTopic = event.newTopic!;
```

Comment:
can use this commented line too

(or could delete it, that's also fine — but definitely no need to have it here commented out when the code is going and referring to `event.newTopic!` anyway)